From e7a287794f86fe16c99de52a5982d7307ae81da9 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 20 Jun 2025 11:11:13 +0200
Subject: [PATCH 881/954] feat(stm32mp2): call stm32mp_board_ddr_power_off in
 power off function

Replace call of ddr_sub_system_clk_off, which only stopped the DDR
sub-system, by stm32mp_board_ddr_power_off() to also deactivate the
associated power supplies.

After this patch the file "plat_ddr.c" is required for BL2 and for BL31.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git de98b419a4b70014277e74d140c5ce3ac5f955d6 feat(stm32mp2): call stm32mp_board_ddr_power_off in power off function]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ib08737cf286b81c39bad64f9b73ac66f5ba24b49
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/355241
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/platform.mk   | 6 +++---
 plat/st/stm32mp2/stm32mp2_pm.c | 3 ++-
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index d74a9de7c3..4ff46d63d8 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -295,7 +295,8 @@ PLAT_BL_COMMON_SOURCES		+=	drivers/st/i2c/stm32_i2c.c
 PLAT_BL_COMMON_SOURCES		+=	plat/st/stm32mp2/stm32mp2_private.c
 
 ifeq ($(STM32MP_M33_TDCID),0)
-PLAT_BL_COMMON_SOURCES		+=	drivers/st/bsec/bsec3.c
+PLAT_BL_COMMON_SOURCES		+=	drivers/st/bsec/bsec3.c \
+					plat/st/stm32mp2/plat_ddr.c
 else
 PLAT_BL_COMMON_SOURCES		+=	plat/st/stm32mp2/stm32mp2_otp.c
 endif
@@ -319,8 +320,7 @@ endif
 BL2_SOURCES			+=	plat/st/stm32mp2/plat_bl2_mem_params_desc.c
 
 BL2_SOURCES			+=	drivers/st/crypto/stm32_hash.c				\
-					plat/st/stm32mp2/bl2_plat_setup.c			\
-					plat/st/stm32mp2/plat_ddr.c
+					plat/st/stm32mp2/bl2_plat_setup.c
 
 BL2_SOURCES			+=	drivers/st/rif/stm32_rifsc.c
 
diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 8a101a5c89..ff17c965c2 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -18,6 +18,7 @@
 #include <drivers/generic_delay_timer.h>
 #include <drivers/st/bsec3_reg.h>
 #include <drivers/st/stm32mp_clkfunc.h>
+#include <drivers/st/stm32mp_ddr.h>
 #include <drivers/st/stm32mp_reset.h>
 #include <drivers/st/stm32mp2_ddr_helpers.h>
 #include <lib/mmio.h>
@@ -1022,7 +1023,7 @@ static void __dead2 stm32_system_off(void)
 #if !STM32MP_M33_TDCID
 	/* Force DDR off */
 	mmio_clrbits_32(rcc_base + RCC_DDRITFCFGR, RCC_DDRITFCFGR_DDRSHR);
-	ddr_sub_system_clk_off();
+	stm32mp_board_ddr_power_off();
 #endif
 
 	/* Prevent interrupts from spuriously waking up this cpu */
-- 
2.53.0

